pkg/cleaner/azure/resourcegroup.go

```diff
@@ -95,3 +96,8 @@ func (c Cleaner) groupHasActivity(ctx context.Context, group resources.Group, si
 	// NotDone returns true when eventIter contains events.
 	return eventIter.NotDone(), nil
 }
+
+// isTerraformCIResourceGroup check if resource group name was created by Terraform CI.
+func isTerraformCIResourceGroup(s string) bool {
+	return strings.HasPrefix(s, "e2e") && len(s) == 7
```

Comment:
I think the length requirement is often a sign of a fragile implementation. Somebody changes the name by appending something and this code here breaks. What characteristics do we have for these names? There should be unit tests for that, or at least a godoc with examples.

Reply:
Not checking for length here is risky IMO. Those name are prefixed with `e2e` which might collide with regular cluster names.
@r7vme Can we update Terraform CI names to have `e2e-` as prefix ?

Reply:
```
e2e70e1
e2e19c5
```

Reply:
That is unfortunate for us here. Does the resource group have any tags or metadata that helps taking more precise decisions?

Reply:
We actually can fix this from giantnetes-terraform side. I think problem was some lenght restrictions in Azure, let me check it and prepare PR.

Reply:
@calvix do you remeber the reason, why we use in azure terraform e2e test cluster names like that?

I've found your change from April like that
```
-CLUSTER=e2e-cluster-$(echo ${CIRCLE_SHA1} | cut -c 1-6)
+CLUSTER=e2e$(echo ${CIRCLE_SHA1} | cut -c 1-4)
```

Reply:
> Does the resource group have any tags or metadata that helps taking more precise decisions?

There is no easily available information. 
* Resource group have a `ManagedBy` property but it is empty in the present case.
* Event in activity logs do not have meaningful informations eitherhttps://portal.azure.com/#@accountsgiantswarm.onmicrosoft.com/resource/subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/e2e70e1/eventlogs

Reply:
Then lets make use of the managed property. Maybe lets have issues for this so it gets addressed and we get back to here later on. It is not really cool but we should then just go with this here for now. But lets make sure we improve this regardless.

Reply:
Either having `e2e-` has name prefix or setting the `ManagedBy` to something like `giantnetes-terraform-ci` both solution works for me.

Reply:
I assume it's Azure storage account name restrictions:

> Storage account name must be between 3 and 24 characters in length and use numbers and lower-case letters only.

PR https://github.com/giantswarm/giantnetes-terraform/pull/138/files